File: app/src/main/java/com/example/strawpoll/network/dtos/StrawpollDTO.kt
package com.example.strawpoll.network.dtos

import com.example.strawpoll.persistence.entities.DatabaseStrawpoll
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class StrawpollDTO(
    val strawpollId:Int,
    val question:String,
    val dateCreated: String,
    val answers:List<AnswerDTO>,
    val alreadyVoted: List<VotedUUIDDTO>
)

fun List<StrawpollDTO>.asDBModel():  List<DatabaseStrawpoll>{
    return map {
        DatabaseStrawpoll(
            id = it.strawpollId,
            question = it.question,
            dateCreated = it.dateCreated
        )
    }
}

File: StrawpollAPI/PollAPI/DTOs/StrawpollDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace PollAPI.DTOs
{
    public class StrawpollDTO
    {
        public int StrawpollId { get; set; }
        public String Question { get; set; }
        public DateTime DateCreated { get; set; }
        public IList<AnswerDTO> Answers { get; set; }
        public IList<VotedUUIDDTO> AlreadyVoted { get; set; }
    }
}
